valuetransfer/src/main/java/nl/tudelft/trustchain/valuetransfer/ValueTransferMainActivity.kt

```diff
@@ -192,14 +195,26 @@ class ValueTransferMainActivity : BaseActivity() {
         /**
          * On initialisation of activity pre-load all fragments to allow instant switching to increase performance
          */
-        fragmentManager.beginTransaction()
-            .add(R.id.container, identityFragment, identityFragmentTag).hide(identityFragment)
-            .add(R.id.container, exchangeFragment, exchangeFragmentTag).hide(exchangeFragment)
-            .add(R.id.container, contactsFragment, contactsFragmentTag).hide(contactsFragment)
-            .add(R.id.container, qrScanController, qrScanControllerTag).hide(qrScanController)
-            .add(R.id.container, settingsFragment, settingsFragmentTag).hide(settingsFragment)
-            .add(R.id.container, walletOverviewFragment, walletOverviewFragmentTag)
-            .commit()
+        if (!requestMoney)
+            fragmentManager.beginTransaction()
+                .add(R.id.container, identityFragment, identityFragmentTag).hide(identityFragment)
```

Comment:
I think you can extract the common elements out of this if-else construct, like:

```
manager = fragmentManager.beginTransaction().hide(...).hide(...)
if(!requestMoney) { manager.hide(...) }
else { manager.hide(...) }
```